src/server/services/procedures/chart/chart.js

```diff
@@ -207,6 +208,22 @@ chart.draw = function(lines, options={}){
         };
     }
 
+    if (options.logscale) {
+        const val = options.logscale;
+        if (typeof val === 'string') opts.logscale = { axes: val, base: 10 };
+        else if (Array.isArray(val)) {
+            const axes = val[0];
+            let base = val[1];
+            if (typeof axes !== 'string') throw Error('logscale axes was not a string (text)');
+            if (base !== undefined) {
+                base = +base;
+                if (!base) throw Error('logscale base was not a number'); // not strictly necessary for gnuplot, but nice for users
```

Comment:
Yeah, I initially allowed negative since gnuplot seems to just take the absolute value internally, but it's probably better to forbid it anyway. Combined with the 'not a number' logic to make the error message clear in all cases (I hadn't noticed before that js makes `+''===0`)

Reply:
Yeah, I agree that it would make more sense to simply disallow nonsensical values (even if allowed by gnuplot). I would also recommend just using BoundedNumber<2> from the input-types.js file so error messages are consistent and we are reusing type parsing code (rather than reinventing the wheel).